### PR TITLE
Reorganized src files to new file structure and naming scheme

### PR DESCRIPTION
**Pull request by flattermann**
As discussed in #orbs-software-dev

### CODE REVIEW

**Comment by flattermann** [comment]
See also #129

**Comment by lyricnz** [comment]
Built, uploaded and tested here. Seems fine - once I moved my config.h to the (new) correct location.

**Review by lyricnz — APPROVED** [review]

**Comment by brett-dot-tech** [comment]
Everything seems to work here but I'm getting some errors in VSC after/during flashing. Seems to be due to the moved config file.
"#include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (/Users/brettturner/Desktop/GitHub/info-orbs/firmware/src/main.cpp)."

I have tried a few high level things like messing around with the platform.ini file, booting from a fresh VSC workspace but no dice. 

Again everything seems to work fine, just keeps throwing the error. Odd.

File: firmware/config/config_helper.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#if !__has_include("config.h")
```
[the remark this answers is not in the document]
Comment (flattermann) [reply firmware/config/config_helper.h:1]: I changed the line with the #error to reflect the new config.h path. That was probably enough for git to not see it as the same file anymore.

File: firmware/src/core/globaltime/GlobalTime.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
-#include "globalTime.h"
+#include "GlobalTime.h"
 
 #include <TimeLib.h>
-#include <config_helper.h>
+#include "config_helper.h"
```
[the remark this answers is not in the document]
Comment (flattermann) [reply firmware/src/core/globaltime/GlobalTime.cpp:4]: I would normally use `""` for our own local files and `<>` for external libraries.
Is that wrong?

[the remark this answers is not in the document]
Comment (flattermann) [reply firmware/src/core/globaltime/GlobalTime.cpp:4]: I know, but I believe using `""` for project files and `<>` for external libraries improves the code's readability and self-explanatory nature:

`#include ""`: Clearly indicates "this is part of our project"
`#include <>`: Clearly indicates "this is a dependency to a system or external library"

This convention aligns with several widely used coding standards, such as the Google C++ Style Guide, and helps distinguish between local project files and external dependencies.

That said, if there's consensus to use `<>` for everything outside the current directory, I'm okay with that too. My main goal is to establish a consistent logic across all files.

[the remark this answers is not in the document]
Comment (lyricnz) [reply firmware/src/core/globaltime/GlobalTime.cpp:4]: FWIW I agree with @flattermann - "" vs <> and what it implies (or at least it does to me)

**Comment by lyricnz** [comment]
NOTE1: This PR changes the capitalisation of a number of file/directory names, which mac/windows don't deal with when doing a git pull/checkout.  You may need to remove local files and do a fresh checkout to make case-sensitive things work properly.

NOTE2: On Mac, in order tto make the squiggly in firmware/lib/OpenFontRender/freetype/config/ftconfig.h go away (which bubbles up to other places), you need to add `$(xcrun --show-sdk-path)/usr/include/AvailabilityMacros.h` to your c++ include-path to make it go away.  This is basically cosmetic, since the ultimate compile is for the board CPU, not your mac.

**Comment by flattermann** [comment]
@lyricnz Oh, thanks, that explains a lot!

Regarding NOTE1: 
On Windows, I could not reproduce the problem with the changed FileName capitalisation.

I can easily switch between
git checkout Code-reorganization
and
git checkout dev
and the files are moved and renamed correctly.

Regarding NOTE2:
I don't have a Mac here, so I can't test it, but can we maybe ship a simplified version of AvailabilityMacros.h with our code?
We probably only need a few #defines in there...

**Comment by flattermann** [comment]
Defining MAC_OS_X_VERSION_MIN_REQUIRED and MAC_OS_X_VERSION_10_4 in our own AvailabilityMacros.h might be enough?

**Pull request merged**